src/main/java/codeu/model/store/basic/MessageStore.java

```diff
@@ -60,10 +63,15 @@ public static MessageStore getTestInstance(PersistentStorageAgent persistentStor
   /** The in-memory list of Messages. */
   private List<Message> messages;
 
+  /** The in-memory list of Messages, organized by author Id */
+  private Map<UUID, List<Message>> messagesByAuthorId;
+
+
   /** This class is a singleton, so its constructor is private. Call getInstance() instead. */
   private MessageStore(PersistentStorageAgent persistentStorageAgent) {
     this.persistentStorageAgent = persistentStorageAgent;
     messages = new ArrayList<>();
+    messagesByAuthorId = new HashMap<>();
   }
```

Comment:
You'll also need to instantiate messagesByAuthorId with some data in loadTestData as well.